src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -169,76 +172,96 @@ public void checkServerTrusted(X509Certificate[] certs, String authType) {
       throw new IllegalStateException("Error while installing the trust manager: " + e.getMessage(), e);
     }
     HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
-    HostnameVerifier allHostsValid = new HostnameVerifier() {
-      public boolean verify(String hostname, SSLSession session) {
-        return true;
-      }
-    };
+    HostnameVerifier allHostsValid = (hostname, session) -> true;
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
 
-  private boolean executeHTTPServiceAndCheckStatusCode() throws IOException {
+  private boolean executeHTTPServiceAndCheckStatusCode() {
     LOG.debug("HTTP Request Attempt No. : {}", ++retryCount);
-    CloseableHttpClient httpClient = createHttpClient(configURL);
 
-    CloseableHttpResponse response = null;
-    try {
+    // Try-with-resources ensures proper resource management
+    try (CloseableHttpClient httpClient = createHttpClient(configURL)) {
       URL url = new URL(configURL);
-      HttpEntityEnclosingRequestBase request = new HttpRequest(URI.create(String.valueOf(url)),
-        config.getMethod());
-
-      if (url.getProtocol().equalsIgnoreCase("https")) {
-        // Disable SSLv3
-        System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
-        if (config.getDisableSSLValidation()) {
-          disableSSLValidation();
-        }
+
+      // Use try-with-resources to ensure response is closed
+      try (CloseableHttpResponse response = executeHttpRequest(httpClient, url)) {
```

Comment:
keep a single try with resources block for httpClient and respose

Reply:
updated